development/tools/jobs/kyma/components_test.go

```diff
@@ -0,0 +1,84 @@
+	{name: "application-connectivity-validator", image: tester.ImageGolangBuildpackLatest},
+	{name: "application-gateway", image: tester.ImageGolangBuildpackLatest},
+	{name: "application-operator", image: tester.ImageGolangBuildpackLatest},
+	{name: "application-registry", image: tester.ImageGolangBuildpackLatest},
+	{name: "asset-metadata-service", image: tester.ImageGolangBuildpack1_11},
+	{name: "asset-store-controller-manager", image: tester.ImageGolangKubebuilder2BuildpackLatest,
+		additionalOptions: []buildjob.Option{
+			buildjob.Since(tester.Release15),
+		},
+	},
+	{name: "asset-store-controller-manager", image: tester.ImageGolangKubebuilderBuildpackLatest,
+		additionalOptions: []buildjob.Option{
+			buildjob.Until(tester.Release14),
+			buildjob.JobFileSuffix("kubebuilder"),
+		},
+	},
+	{name: "asset-upload-service", image: tester.ImageGolangBuildpack1_11},
+	{name: "cms-controller-manager", image: tester.ImageGolangKubebuilder2BuildpackLatest,
+		additionalOptions: []buildjob.Option{
+			buildjob.Since(tester.Release15),
+		},
+	},
+	{name: "cms-controller-manager", image: tester.ImageGolangKubebuilderBuildpackLatest,
+		additionalOptions: []buildjob.Option{
+			buildjob.Until(tester.Release14),
+			buildjob.JobFileSuffix("kubebuilder"),
+		},
+	},
+	{name: "cms-services", image: tester.ImageGolangBuildpack1_12},
+	{name: "compass-runtime-agent", image: tester.ImageGolangBuildpack1_11},
+	{name: "connection-token-handler", image: tester.ImageGolangBuildpackLatest},
+	{name: "connectivity-certs-controller", image: tester.ImageGolangBuildpackLatest},
+	{name: "connector-service", image: tester.ImageGolangBuildpackLatest},
+	{name: "console-backend-service", image: tester.ImageGolangBuildpack1_11},
+	{name: "dex-static-user-configurer", image: tester.ImageBootstrapLatest},
+	{name: "etcd-tls-setup-job", image: tester.ImageGolangBuildpack1_11},
+	{name: "event-bus", image: tester.ImageGolangBuildpack1_11},
+	{name: "event-service", image: tester.ImageGolangBuildpack1_11},
+	{name: "helm-broker", image: tester.ImageGolangKubebuilderBuildpackLatest,
+		additionalOptions: []buildjob.Option{
+			buildjob.Until(tester.Release14),
+			buildjob.JobFileSuffix("deprecated"),
+		},
+	},
+	{name: "iam-kubeconfig-service", image: tester.ImageGolangBuildpack1_12},
+	{name: "istio-kyma-patch", image: tester.ImageBootstrapLatest},
+	{name: "k8s-dashboard-proxy", image: tester.ImageGolangBuildpack1_11},
+	{name: "function-controller", image: tester.ImageGolangKubebuilderBuildpackLatest},
+	{name: "kubeless-images/nodejs", image: tester.ImageGolangBuildpack1_11},
+	{name: "kyma-operator", image: tester.ImageGolangBuildpack1_12},
+	{name: "namespace-controller", image: tester.ImageGolangBuildpackLatest},
+	{name: "service-binding-usage-controller", image: tester.ImageGolangBuildpack1_11},
+	{name: "xip-patch", image: tester.ImageBootstrapLatest},
+}
+
+func TestComponentJobs(t *testing.T) {
```

Comment:
**My personal opinion**

I do not like magic :) 

having sth like `buildjob.NewSuite(opts...).Run(t)` is to magic. 
there is a lot of if statement inside. 

I will prefer to have more descriptive API and split it to sth like
```go
buildjob.NewPresubmitJobSuite(opts...).Run(t)
buildjob.NewPostsubmitJobSuite(opts...).Run(t)
buildjob.NewReleaseJobSuite(opts...).Run(t)

// or
suite := buildjob.NewSuite(opts...)
suite.TestPresubmitJobs(t)
suite.TestPostsubmitJobs(t)
suite.TestReleaseJobs(t)
```

Reply:
I wanted to avoid that because components shouldn't differ and should be all tested the same way. The more commands need to be executed the harder it is to achieve. 

The current API reflects the build logic, not the Prow logic. So you rather define a component and its specific configuration, rather than test prow jobs directly.